unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs

```diff
@@ -5124,106 +5092,99 @@ private IEnumerator checkDropHandObjectActionFast(SimObjPhysics currentHandSimOb
+        // sim object that they target.
+        private SimObjPhysics getTargetObject(
+                string objectId,
+                bool forceAction = false
+        ) {
+            // an objectId was given, so find that target in the scene if it exists
+            if (!physicsSceneManager.ObjectIdToSimObjPhysics.ContainsKey(objectId)) {
+                errorMessage = "Object ID appears to be invalid.";
+                return null;
             }
 
-            //an objectId was given, so find that target in the scene if it exists
-            else
-            {
-                if (!physicsSceneManager.ObjectIdToSimObjPhysics.ContainsKey(action.objectId)) {
-                    errorMessage = "Object ID appears to be invalid.";
-                    actionFinished(false);
-                    return;
-                }
-                
-                //if object is in the scene and visible, assign it to 'target'
-                foreach (SimObjPhysics sop in VisibleSimObjs(action)) 
-                {
-                    target = sop;
-                }
+            // if object is in the scene and visible, assign it to 'target'
+            SimObjPhysics target = null;
+            foreach (SimObjPhysics sop in VisibleSimObjs(objectId: objectId, forceVisible: forceAction)) {
+                target = sop;
             }
+            return target;
+        }
 
-            if (target) {
-
-                if (!action.forceAction && target.isInteractable == false) {
-                    errorMessage = "object is visible but occluded by something: " + action.objectId;
-                    actionFinished(false);
-                }
-
-                if (target.GetComponent<CanOpen_Object>()) {
-                    CanOpen_Object codd = target.GetComponent<CanOpen_Object>();
-
-                    //if object is open, close it
-                    if (codd.isOpen) {
-                        // codd.Interact();
-                        // actionFinished(true);
-                        StartCoroutine(InteractAndWait(codd, action.simplifyPhysics));
-                    } else {
-                        errorMessage = "object already closed: " + action.objectId;
-                        actionFinished(false);
-                    }
-                }
+        // Helper method that parses (x and y) parameters to return the
+        // sim object that they target.
+        private SimObjPhysics getTargetObject(
+            float x,
+            float y,
+            bool forceAction
+        ) {
+            // no target object specified, so instead try and use x/y screen coordinates
+            SimObjPhysics target = null;
+            if(!ScreenToWorldTarget((float) x, (float) y, ref target, !forceAction)) {
+                // error message is set insice ScreenToWorldTarget
+                return null;
+            }
+            return target;
+        }
```

Comment:
Eventually `getTargetObject` will be used for every interaction action that targets an object by either `x/y` or `objectId`. Currently, most actions repeat this code in exactly the same manner. Adding this to every action, however, will be a future PR.

Note that above this, it is overloaded to also take in an `objectId`.

Reply:
Looks good. Should be a pretty straightforward future PR to swap this in to the rest of the actions.